### DISCUSSION TITLE
Test optimized or debug builds?

### DISCUSSION THREAD

**Discussion by sliekens** [xunit/xunit#2589] [Question] [answered]
Hi,

Google couldn't find what I was looking for so I figure nobody has asked the question before. Should I compile my SUT and test assemblies using compiler optimizations, or in other words `dotnet test --configuration Release`? Why or why not? Assuming all else being equal, I'm not using ifdefs or anything like that.

I imagine it depends on the environment (command line vs. UI test runner, local machine vs. CI server). I'm fairly certain I've seen CI workflows that use a build matrix to test multiple configs.

What do you think?

**Comment by bradwilson** [comment, marked as answer]
For CI builds, we always test `Release`. While developing, I usually test `Debug` (so that I can easily debug the test as needed), but the command line build script that we use (that I always run before pushing things up) mimics the CI build behavior, so it will run tests in `Release` for me locally as well.
